carto/auth.py

```diff
@@ -115,3 +109,64 @@ def get_user_name(self, base_url):
                                    "'https://{user_name}.carto.com', " +
                                    "'https://carto.com/user/{user_name}' " +
                                    "or a similar one based on your domain"))
+
+    def prepare_send(self, http_method, **requests_args):
+        http_method = http_method.lower()
+        if (http_method in ['post', 'put']) and "json" in requests_args:
+            requests_args["json"].update({"api_key": self.api_key})
+        else:
+            if "params" not in requests_args:
+                requests_args["params"] = {}
+            requests_args["params"].update({"api_key": self.api_key})
+
+        return http_method, requests_args
+
+
+class NonVerifiedAPIKeyAuthClient(APIKeyAuthClient):
+    """
+    This class provides you with authenticated access to CARTO's APIs using
+    your API key but avoids verifying SSL certificates. This is useful for onpremises instances of CARTO
+
+    You can find your API key by clicking on the API key section of the user
+    dropdown menu
+    """
+    def __init__(self, base_url, api_key, organization=None, session=None):
+        """
+        Init method
+
+        :param base_url: Base URL. API endpoint paths will always be relative
+        to this URL
+        :param api_key: API key
+        :param organization: For enterprise users, organization user belongs to
+        :param session: requests' session object
+        :type api_key: str
+        :type organization: str
+
+        :return:
+        """
+        super(NonVerifiedAPIKeyAuthClient, self).__init__(base_url, api_key, organization, session)
+
+    def send(self, relative_path, http_method, **requests_args):
+        """
+        Makes an API-key-authorized request not verifying SSL certs
+
+        :param relative_path: URL path relative to self.base_url
+        :param http_method: HTTP method
+        :param requests_args: kwargs to be sent to requests
+        :type relative_path: str
+        :type http_method: str
+        :type requests_args: kwargs
+
+        :return:
+            A request response object
+        :raise:
+            CartoException
+        """
+        try:
+            http_method, requests_args = self.prepare_send(http_method, **requests_args)
+            requests_args["verify"] = False
+            return super(APIKeyAuthClient, self).send(relative_path,
```

Comment:
why can't you just call `super(NonVerifiedAPIKeyAuthClient, self).send()` with the augmented `request_args` directly, without even doing `prepare_send()`?

Reply:
The reason is because `NonVerifiedAPIKeyAuthClient` is meant to be an utility class so that the developer does not need to know that `verify=False` is needed for avoiding SSL verification.

Anyway, I found out that [we already supported non-verified SSL calls](https://github.com/CartoDB/carto-python/issues/59#issuecomment-316439017)